framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -923,6 +1066,50 @@ private AnnotationMirror createNumberAnnotationMirror(List<Number> values) {
                 "ValueAnnotatedTypeFactory: unexpected class: " + first.getClass());
     }
 
+    public AnnotationMirror createIntRangeAnnotation(Range range) {
```

Comment:
Yes, I meant that the first line of every routine like `union` should test against `NOTHING`.  (Sorry that wasn't clear.)

It could be something like this:

```java
    public Range union(Range right) {
        if ((this == NOTHING) || (right == NOTHING)) return NOTHING;
        long resultFrom = Math.min(from, right.from);
        long resultTo = Math.max(to, right.to);
        return new Range(resultFrom, resultTo);
    }
```